src/modules/hip/kernel/slice.hpp

```diff
@@ -32,77 +87,441 @@ __global__ void slice_ncdhw_hip_tensor(T *srcPtr,
+            }
+            else if (needPadding && dstGenericDescPtr->layout == RpptLayout::NDHWC)
+            {
+                hipLaunchKernelGGL(fill_value_ndhwc_hip_tensor,
+                                   dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                                   dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                                   0,
+                                   handle.GetStream(),
+                                   dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
+                                   make_uint2(dstGenericDescPtr->strides[1], dstGenericDescPtr->strides[2]),
+                                   make_uint3(maxDepth, maxHeight, maxWidth),
+                                   fillValue);
+            }
+        }
+    }
+    else if(numDims == 3)
+    {
+        // set the dimsOrder and globalthreads values required for NHWC layout
+        Rpp32s dimsOrder[2] = {0, 1};
+        int globalThreads_x = (dstGenericDescPtr->strides[1] / 3 + 7) >> 3; // W - width  (x direction) - vectorized for 8 element loads/stores per channel
+        int globalThreads_y = dstGenericDescPtr->dims[1];                   // H - height (y direction)
+        int globalThreads_z = 1;
+
+        // change the dimsOrder and globalthreads values if layout is NCHW
+        if((srcGenericDescPtr->layout == RpptLayout::NCHW) && (dstGenericDescPtr->layout == RpptLayout::NCHW))
+        {
+            dimsOrder[0] = 1;  // height
+            dimsOrder[1] = 2;  // width
+            int globalThreads_x = (dstGenericDescPtr->strides[2] + 7) >> 3; // W - width  (x direction) - vectorized for 8 element loads/stores per channel
+            int globalThreads_y = dstGenericDescPtr->dims[2];               // H - height (y direction)
+            int globalThreads_z = 1;
+        }
+
+        for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
+        {
+            Rpp32s *anchor = &anchorTensor[batchCount * numDims];
+            Rpp32s *shape = &shapeTensor[batchCount * numDims];
+            Rpp32u *roi = roiTensor + batchCount * numDims * 2;
+            Rpp32s *length = reinterpret_cast<Rpp32s *>(&roi[numDims]);
+            Rpp32u maxHeight = std::min(shape[dimsOrder[0]], length[dimsOrder[0]] - anchor[dimsOrder[0]]);
+            Rpp32u maxWidth = std::min(shape[dimsOrder[1]], length[dimsOrder[1]] - anchor[dimsOrder[1]]);
+
+            // check if padding is needed
+            bool needPadding = (((anchor[dimsOrder[0]] + shape[dimsOrder[0]]) > length[dimsOrder[0]]) ||
+                                ((anchor[dimsOrder[1]] + shape[dimsOrder[1]]) > length[dimsOrder[1]]));
+
+            // launch kernel for filling the padded region with fill value specified
+            if (needPadding && dstGenericDescPtr->layout == RpptLayout::NCHW)
```

Comment:
Like this condition, pick either src everywhere or dst everywhere. Since we know src layout = dst layout for this function.